### PR TITLE
AMQP-558: amqp-client 3.6.0

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/AMQP-558

Minimum supported amqp-client is now 3.4.0.

__do not merge until 3.6.0 is in maven central/spring repo__

### CODE REVIEW

**Comment by artembilan** [comment]
> do not merge until 3.6.0 is in maven central/spring repo

Do you have already some news when it is planned for that?

CC @michaelklishin

**Comment by garyrussell** [comment]
@artembilan I didn't bother Michael with this because of the comment on the web site...

>We attempt to upload new versions of the Java client on the day of release; however the Maven servers are sometimes unavailable, so there may be a delay of a few days between a new release and its appearance in the central Maven repository. Please be patient.

**Comment by michaelklishin** [comment]
We've reworked our entire build system in 3.6.0 and publishing to `oss.sonatype.org` didn't work as expected. Sadly, it's a thing you learn when you actually do a release. We'll publish Java client artefacts ASAP.

**Comment by artembilan** [comment]
Thank you @michaelklishin !

No rush.

Merry Christmas and Happy New Year!
Hope at least you have a snow there in Russia :smile:

**Comment by michaelklishin** [comment]
Likewise!

It's been well above 0 degrees C for a good week in Moscow. I am glad I'm not into outdoor winter sports ;)

File: build.gradle [hunk 1]
```diff
@@ -87,10 +88,10 @@ subprojects { subproject ->
 		log4jVersion = '1.2.17'
 		logbackVersion = '1.1.2'
 		mockitoVersion = '1.9.5'
-		rabbitmqVersion = project.hasProperty('rabbitmqVersion') ? project.rabbitmqVersion : '3.5.7'
+		rabbitmqVersion = project.hasProperty('rabbitmqVersion') ? project.rabbitmqVersion : '3.6.0'
 		rabbitmqHttpClientVersion = '1.0.0.RELEASE'
 
-		springVersion = project.hasProperty('springVersion') ? project.springVersion : '4.2.2.RELEASE'
+		springVersion = project.hasProperty('springVersion') ? project.springVersion : '4.2.4.RELEASE'
```
Comment (artembilan) [inline build.gradle]: The `TODO: when Spring 4.2.4 is the minimum Spring Framework version, use encodeToUrlSafeString() SPR-13784.` in the `AnonymousQueue` can be addressed with this PR.

**Comment by artembilan** [comment]
Looks like the `3.6.0` is in the Maven Central: http://search.maven.org/#artifactdetails|com.rabbitmq|amqp-client|3.6.0.
Merging addressing the `TODO` concern above.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/5c6e4818d3b7125a391d4b2724957c83b0181ed8

File: build.gradle (hunk 1 above)
Comment (garyrussell) [inline build.gradle]: I was not planning on making 4.2.4 the minimum for 1.6; it seems premature to me - especially for such a trivial thing.

See http://docs.spring.io/spring-amqp/docs/1.6.0.BUILD-SNAPSHOT/reference/html/_introduction.html#compatibility

Comment (artembilan) [inline build.gradle]: I think the upgrade isn't a big deal :smile:  http://spring.io/blog/2015/12/17/spring-framework-4-2-4-4-1-9-released

Comment (artembilan) [inline build.gradle]: Although with that we should update that sentence in the Manual...

Comment (garyrussell) [inline build.gradle]: I have no problem making 4.2.x the minimum required version if there's some compelling reason but I don't feel this is compelling.

Comment (artembilan) [inline build.gradle]: Well, then I misunderstood the TODO a bit:
https://github.com/spring-projects/spring-amqp/commit/5c6e4818d3b7125a391d4b2724957c83b0181ed8#diff-76a536976554759111fcf435404171aaL115
...

Comment (garyrussell) [inline build.gradle]: Yes, you misunderstood - I meant there that when S-AMQP __requires__ 4.2.x not when it pulls it in by default.

Comment (artembilan) [inline build.gradle]: Not so obvious though, but that isn't a big deal to revert.
Thank you for pointing that out!

**Pull request closed without merging**